test/unified_format.py

```diff
@@ -1103,15 +1102,21 @@ def _testOperation_loop(self, spec):
                 if iteration_key:
                     self.entity_map._entities[iteration_key] += 1
             except AssertionError as exc:
+                if iteration_key:
+                    self.entity_map._entities[iteration_key] += 1
                 if failure_key or error_key:
                     self.entity_map[failure_key or error_key].append({
-                        "error": exc, "time": time.time()})
+                        "error": str(exc), "time": time.time(), "type": type(
+                            exc)})
                 else:
                     raise exc
             except Exception as exc:
+                if iteration_key:
+                    self.entity_map._entities[iteration_key] += 1
```

Comment:
If we want to increment iteration_key every time it would be simpler to do it _before_  at the top of the loop rather than in every case.

Reply:
Done.